File: examples/example-57/js/features/step_definitions/webdriver.js
const { Given, When, Then } = require('@cucumber/cucumber');
const { By, until } = require('selenium-webdriver');
const assert = require('assert');

Given('the homepage is opened with webdriver', async function () {
    await this.driver.get('http://localhost:9999');
});

When('{string} is typed into the search box with webdriver', async function (text) {
    const input = await this.driver.findElement(By.css('input[type="text"]'));
    await input.sendKeys(text);
});

When('the {string} button is pressed with webdriver', async function (text) {
    const button = await this.driver.findElement(
        By.css(`input[type="submit"][value="${text}"]`));
    await button.click();
});

Then('the results should display {string} with webdriver', async function (text) {
    const results = await this.driver.wait(
        until.elementLocated(By.css('div#results')), 10000);
    const resultsText = await results.getText();
    assert.ok(resultsText.includes(text));
});

File: examples/example-60/ruby/features/step_definitions/webdriver.rb
require 'rspec'

Given('the homepage is opened with webdriver') do
    @driver.navigate.to 'http://host.docker.internal:9999'
end

When('{string} is typed into the search box with webdriver') do |value|
    wait = Selenium::WebDriver::Wait.new(timeout: 10)
    wait.until { @driver.find_element(id: 'search_box').displayed? }
    element = @driver.find_element(id: 'search_box')
    element.send_keys(value)
end

When('the {string} button is pressed with webdriver') do |btn|
    path = "//input[@type='submit'][@value='#{btn}']"
    wait = Selenium::WebDriver::Wait.new(timeout: 10)
    wait.until { @driver.find_element(xpath: path).displayed? }
    button = @driver.find_element(xpath: path)
    button.click
end

Then('the results should display {string} with webdriver') do |results|
    wait = Selenium::WebDriver::Wait.new(timeout: 10)
    wait.until { @driver.find_element(id: 'results').displayed? }
    text = @driver.find_element(id: 'results').text
    expect(text).to eq(results)
end